dotCMS/src/integration-test/java/com/dotcms/rendering/velocity/services/HTMLPageAssetRenderedTest.java

```diff
@@ -1298,6 +1300,72 @@ public void shouldRenderUsingOtherSiteContainerAndNotAdvanceTemplate() throws Ex
         Assert.assertTrue(html.contains("content2content1"));
     }
 
+    //Data Provider for the Widget Pre-execute code test
+    @DataProvider
+    public static Object[] dataProviderWidgetPreExecuteCode() {
+        return new Object[] {
+                new WidgetPreExecuteCodeTestCase(PageMode.EDIT_MODE),
+                new WidgetPreExecuteCodeTestCase(PageMode.PREVIEW_MODE),
+                new WidgetPreExecuteCodeTestCase(PageMode.LIVE),
+        };
+    }
+
+    private static class WidgetPreExecuteCodeTestCase {
+        final PageMode pageMode;
+
+        public WidgetPreExecuteCodeTestCase(final PageMode pageMode) {
+            this.pageMode = pageMode;
+        }
+    }
+
+    @Test
+    @UseDataProvider("dataProviderWidgetPreExecuteCode")
+    public void test_WidgetPreExecuteCodeShowRegardlessPageMode(final WidgetPreExecuteCodeTestCase testCase) throws Exception {
+
+        //Update the Preexecute Field to have some code in it
+        final String preExcuteCode = "PreExecute Code Displayed";
+        ContentType contentType = TestDataUtils.getWidgetLikeContentType();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentType' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5178716)